tests/test_kernels.py

```diff
@@ -700,6 +700,74 @@ def test_connectivities_key_kernel(self, adata: AnnData):
         np.testing.assert_array_equal(T_cr.A, adata.obsp[key])
 
 
+class TestVelocityKernelReadData:
+    def test_read_correct_from_layers(self, adata: AnnData):
+        xkey = "Ms"
+        vkey = "velocity"
+        gene_subset = adata.var[f"{vkey}_genes"]
+        nans_v = np.isnan(np.sum(adata.layers[vkey], axis=0))
+        gene_subset_reduced = adata.var[f"{vkey}_genes"].copy()
+        gene_subset_reduced[10:] = False
+
+        vk = VelocityKernel(
+            adata,
+            xkey=xkey,
+            vkey=vkey,
+            attr="layers",
+            gene_subset=None,
+        )
+        assert np.all(
+            vk._xdata == adata.layers[xkey][:, np.asarray(gene_subset) & ~nans_v]
+        )
+        assert np.all(
+            vk._vdata == adata.layers[vkey][:, np.asarray(gene_subset) & ~nans_v]
+        )
+
+        vk_red = VelocityKernel(
+            adata,
+            xkey=xkey,
+            vkey=vkey,
+            attr="layers",
+            gene_subset=gene_subset_reduced,
+        )
+
+        assert np.all(
+            vk_red._vdata
+            == adata.layers[vkey][:, np.asarray(gene_subset_reduced) & ~nans_v]
+        )
+        assert np.all(
+            vk_red._xdata
+            == adata.layers[xkey][:, np.asarray(gene_subset_reduced) & ~nans_v]
+        )
+
+    def test_read_correct_from_obsm(self, adata: AnnData):
```

Comment:
This test doesn't need to exist; the previous one should be parametrized using `@pytest.mark.parametrize("attr", ["layers", "obsm"])`